### DISCUSSION TITLE
Table key must be non-empty string in `AMQPTable`

### DISCUSSION THREAD

**Discussion by venugopal8** [php-amqplib/php-amqplib#1117] [Help wanted]
Facing an issue while consuming messages from the queue. There are 20+ messages in my queue while reading getting below error:

```
PHP Fatal error:  Uncaught PhpAmqpLib\Exception\AMQPInvalidArgumentException: Table key must be non-empty string up to 128 chars in length in /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPTable.php:33
Stack trace:
#0 /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(293): PhpAmqpLib\Wire\AMQPTable->set()
#1 /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(304): PhpAmqpLib\Wire\AMQPReader->read_table()
#2 /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Message/AMQPMessage.php(480): PhpAmqpLib\Wire\AMQPReader->read_table_object()
#3 /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(288): PhpAmqpLib\Message\AMQPMessage->load_properties()
#4 /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(277): PhpAmqpLib\Channel\AbstractChannel->createMessage()
#5 /vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.p in /var/www/html/MW/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPTable.php on line 33
```

When I checked by reading a message from Management UI I got the header as an empty value.

**Comment by lukebakken** [comment]
@ramunasd this is from an internal discussion at VMware.

@venugopal8 - Can you explain **how** a message was published with an empty key field in the message headers? Can this be easily reproduced?

**Reply by venugopal8** [reply]
Publishing a message will be done by some other team. Our application will consume a message from the queue to process further. 
Since we don't have control of this reproducing the scenario is difficult. I have informed the Publishing team of the same error since they were waiting for the process to be completed and the message was stuck in the queue.
We were not facing this issue earlier. Now only we are getting this issue.

Tried to reproduce this locally but the header is empty even though I can consume the message with the same code base and not get any exceptions. This exception is faced on the server.  
Can you please guide me on the use case and reason for getting this exception `Uncaught PhpAmqpLib\Exception\AMQPInvalidArgumentException`

**Reply by lukebakken** [reply]
> We were not facing this issue earlier. Now only we are getting this issue.

What does "earlier" mean? What has changed since then? Did the publishing application change? Was `php-amqplib` upgraded in consumers?

**Reply by venugopal8** [reply]
I am not aware of the publishing side. But we have upgraded `php-amqplib` and moved from AMQP to AMQPS by upgrading RabbitMQ 3.11.9

**Reply by lukebakken** [reply]
What are the previous and current versions of `php-amqplib`? That will help me determine what changed to cause this issue, since we don't have a way to reproduce it.

**Reply by venugopal8** [reply]
I have upgraded the php-amqplib to 3.5
I am able to reproduce this issue locally. While posting a message if I add multiple spaces with semicolon in header like  `        :          `
The above value is causing the issue. Is there any way to handle this on the consumer end?

**Reply by venugopal8** [reply]
Due to this invalid header, the other valid messages are been stuck in the queue.  
Can you please guide me, to unblock the valid messages.
Tried moving the invalid to DLQ from the consumer end but was not able to move.

**Reply by lukebakken** [reply]
> I am able to reproduce this issue locally. While posting a message if I add multiple spaces with semicolon in header like        :         

Please please please share code I can run to do exactly the same thing you are! That removes **ALL** of the guessing I have to do.

Share the code via a git repository I can clone. I should only have to run `composer install` then `php ./repro.php` to run it. Assume I have RabbitMQ running on `localhost:5672`
